### PR TITLE
Bael 8020 finding min max

### PR DESCRIPTION
**Pull request by sagaofsilence**

### CODE REVIEW

File: core-java-modules/core-java-streams-simple/src/main/java/com/baeldung/streams/minmaxbygroup/OrderItem.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
+package com.baeldung.streams.minmaxbygroup;
+
+public class OrderItem {
+  private Long id;
+  private Double price;
+  private OrderItemCategory category;
+
+  // other fields
```
Comment (vinipx) [inline core-java-modules/core-java-streams-simple/src/main/java/com/baeldung/streams/minmaxbygroup/OrderItem.java:8]: let's remove the comment - not necessary

File: core-java-modules/core-java-streams-simple/src/main/java/com/baeldung/streams/minmaxbygroup/OrderProcessor.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
+package com.baeldung.streams.minmaxbygroup;
+
+import java.util.DoubleSummaryStatistics;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import org.apache.commons.lang3.tuple.Pair;
+
+/** The type Order processor. */
```
Comment (vinipx) [inline core-java-modules/core-java-streams-simple/src/main/java/com/baeldung/streams/minmaxbygroup/OrderProcessor.java:9]: let's remove the comment - not necessary

File: core-java-modules/core-java-streams-simple/src/main/java/com/baeldung/streams/minmaxbygroup/OrderProcessor.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+public class OrderProcessor {
+  /**
+   * Group by category with min max per group.
```
Comment (vinipx) [inline core-java-modules/core-java-streams-simple/src/main/java/com/baeldung/streams/minmaxbygroup/OrderProcessor.java:12]: let's remove the comments - not necessary

File: core-java-modules/core-java-streams-simple/src/main/java/com/baeldung/streams/minmaxbygroup/OrderProcessor.java [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+   *
+   * @param orderItems the order items
+   * @return the grouped items
+   */
+  public Map<OrderItemCategory, Pair<Double, Double>> groupByCategoryWithMinMax(
+      List<OrderItem> orderItems) {
+    Map<OrderItemCategory, DoubleSummaryStatistics> categoryStatistics =
+        orderItems.stream()
```
Comment (vinipx) [inline core-java-modules/core-java-streams-simple/src/main/java/com/baeldung/streams/minmaxbygroup/OrderProcessor.java:20]: you can keep the orderItems.stream() in the same line as variable declaration

File: core-java-modules/core-java-streams-simple/src/test/java/com/baeldung/streams/minmaxbygroup/OrderProcessorUnitTest.java [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import org.apache.commons.lang3.tuple.Pair;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+class OrderProcessorUnitTest {
+  @Test
```
Comment (vinipx) [inline core-java-modules/core-java-streams-simple/src/test/java/com/baeldung/streams/minmaxbygroup/OrderProcessorUnitTest.java:11]: let's break the line here between the class and Test annotation

File: core-java-modules/core-java-streams-simple/src/test/java/com/baeldung/streams/minmaxbygroup/OrderProcessorUnitTest.java [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+            new OrderItem(4L, OrderItemCategory.FURNITURE, 200.20),
+            new OrderItem(5L, OrderItemCategory.FURNITURE, 215.00),
+            new OrderItem(6L, OrderItemCategory.CLOTHING, 50.75),
+            new OrderItem(7L, OrderItemCategory.CLOTHING, 75.00),
+            new OrderItem(8L, OrderItemCategory.CLOTHING, 75.00));
+
+    OrderProcessor orderProcessor = new OrderProcessor();
+    final Map<OrderItemCategory, Pair<Double, Double>> orderItemCategoryPairMap =
+        orderProcessor.groupByCategoryWithMinMax(items);
+    Assertions.assertEquals(
```
Comment (vinipx) [inline core-java-modules/core-java-streams-simple/src/test/java/com/baeldung/streams/minmaxbygroup/OrderProcessorUnitTest.java:28]: you can import static the Assertions and leave each assertEquals in one single line

**Review by vinipx — CHANGES_REQUESTED** [review]

File: core-java-modules/core-java-streams-simple/src/main/java/com/baeldung/streams/minmaxbygroup/OrderItem.java (hunk 1 above)
Comment (sagaofsilence) [reply core-java-modules/core-java-streams-simple/src/main/java/com/baeldung/streams/minmaxbygroup/OrderItem.java:8]: Done

File: core-java-modules/core-java-streams-simple/src/main/java/com/baeldung/streams/minmaxbygroup/OrderProcessor.java (hunk 2 above)
Comment (sagaofsilence) [reply core-java-modules/core-java-streams-simple/src/main/java/com/baeldung/streams/minmaxbygroup/OrderProcessor.java:9]: Done

File: core-java-modules/core-java-streams-simple/src/main/java/com/baeldung/streams/minmaxbygroup/OrderProcessor.java (hunk 3 above)
Comment (sagaofsilence) [reply core-java-modules/core-java-streams-simple/src/main/java/com/baeldung/streams/minmaxbygroup/OrderProcessor.java:12]: Done.

File: core-java-modules/core-java-streams-simple/src/main/java/com/baeldung/streams/minmaxbygroup/OrderProcessor.java (hunk 4 above)
Comment (sagaofsilence) [reply core-java-modules/core-java-streams-simple/src/main/java/com/baeldung/streams/minmaxbygroup/OrderProcessor.java:20]: I had to format that manually. I was using `intellij-baeldung-style.xml`, then the formatting takes it to next line.

File: core-java-modules/core-java-streams-simple/src/test/java/com/baeldung/streams/minmaxbygroup/OrderProcessorUnitTest.java (hunk 5 above)
Comment (sagaofsilence) [reply core-java-modules/core-java-streams-simple/src/test/java/com/baeldung/streams/minmaxbygroup/OrderProcessorUnitTest.java:11]: Done

File: core-java-modules/core-java-streams-simple/src/test/java/com/baeldung/streams/minmaxbygroup/OrderProcessorUnitTest.java (hunk 6 above)
Comment (sagaofsilence) [reply core-java-modules/core-java-streams-simple/src/test/java/com/baeldung/streams/minmaxbygroup/OrderProcessorUnitTest.java:28]: Done. I had to format that manually. I was using intellij-baeldung-style.xml, then the formatting takes it to next line.

**Comment by sagaofsilence** [comment]
Changes done.

**Pull request merged**